File: SpiderEngine/Model/SpiderPayload.cs
namespace SpiderEngine.Model;

public class SpiderPayload
{
    public SpiderPayloadType Type { get; set; } = default!;
    public string Patten { get; set; } = default!;
    public IEnumerable<SpiderKeyValuePair> Value { get; set; } = default!;
    public IEnumerable<SpiderKeyValuePair>? Header { get; set; } = default!;
}

public enum SpiderPayloadType
{
    Text,
    Json,
    Form,
    Params,
}

File: android/src/main/kotlin/com/itoken/team/wutspider/model/SpiderPayload.kt
package com.itoken.team.wutspider.model

import kotlinx.serialization.ExperimentalSerializationApi
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonNames

@Serializable
data class SpiderPayload @OptIn(ExperimentalSerializationApi::class) constructor(
    val type: SpiderPayloadType = SpiderPayloadType.TEXT,
    @JsonNames("patten") val pattern: String? = null,
    val value: List<SpiderKeyValuePair>? = null,
    val header: List<SpiderKeyValuePair>? = null
)


enum class SpiderPayloadType {
    @SerialName("text") TEXT,
    @SerialName("json") JSON,
    @SerialName("form") FORM,
    @SerialName("params") PARAMS
}